consensus/ssz_derive/src/lib.rs

```diff
@@ -152,6 +153,72 @@ pub fn ssz_encode_derive(input: TokenStream) -> TokenStream {
     output.into()
 }
 
+/// Derive `Encode` for a restricted subset of all possible enum types.
+///
+/// Only supports:
+/// - Enums with a single field per variant, where
+///     - All fields are variably sized from an SSZ-perspective (not fixed size).
```

Comment:
Out of interest, is there a reason (apart from simplicity) why they must all be variable-sized rather than consistently variable/fixed-size?

Reply:
Hmm perhaps they don't even need to be consistent :thinking:

Reply:
> Hmm perhaps they don't even need to be consistent thinking

Oh no, `Encode::is_ssz_fixed_len(self)` won't allow for this.

Reply:
> Out of interest, is there a reason (apart from simplicity) why they must all be variable-sized rather than consistently variable/fixed-size?

Just simplicity. I figured all of our enums would be for variable-sized `superstructs`. I don't think it would be worth adding support for enums with equal fixed-size variants before we have some use for them (and I doubt we would).